From 42f30eb3cc2ffdf352829f6b3bf5e176d699a1a5 Mon Sep 17 00:00:00 2001
From: Kevin Cernekee <cernekee@gmail.com>
Date: Wed, 5 Mar 2014 19:56:42 -0800
Subject: [PATCH] gl: stdint, read-file: fix missing SIZE_MAX on Android

commit cb3c90598c13d3db616c0c1e62c5d59ed80b069d upstream (gnulib).

This is basically one of the options Bruno Haible proposed in:
http://lists.gnu.org/archive/html/bug-gnulib/2012-01/msg00282.html
* lib/sys_types.in.h (_GL_INCLUDING_UNISTD_H): New macro.
* lib/stdint.in.h: Use it.
* modules/stdint (Depends-on): Add sys_types.
---
 gl/stdint.in.h    |    3 +--
 gl/sys_types.in.h |    2 ++
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/gl/stdint.in.h b/gl/stdint.in.h
index 07a4b80..a109bdd 100644
--- a/gl/stdint.in.h
+++ b/gl/stdint.in.h
@@ -38,8 +38,7 @@
    other system header files; just include the system's <stdint.h>.
    Ideally we should test __BIONIC__ here, but it is only defined after
    <sys/cdefs.h> has been included; hence test __ANDROID__ instead.  */
-#if defined __ANDROID__ \
-    && defined _SYS_TYPES_H_ && !defined __need_size_t
+#if defined __ANDROID__ && defined _GL_INCLUDING_SYS_TYPES_H
 # @INCLUDE_NEXT@ @NEXT_STDINT_H@
 #else
 
diff --git a/gl/sys_types.in.h b/gl/sys_types.in.h
index 689ce50..bd7cb32 100644
--- a/gl/sys_types.in.h
+++ b/gl/sys_types.in.h
@@ -23,7 +23,9 @@
 #ifndef _@GUARD_PREFIX@_SYS_TYPES_H
 
 /* The include_next requires a split double-inclusion guard.  */
+# define _GL_INCLUDING_SYS_TYPES_H
 #@INCLUDE_NEXT@ @NEXT_SYS_TYPES_H@
+# undef _GL_INCLUDING_SYS_TYPES_H
 
 #ifndef _@GUARD_PREFIX@_SYS_TYPES_H
 #define _@GUARD_PREFIX@_SYS_TYPES_H
-- 
1.7.9.5

